### PR TITLE
DataNode: Always using the hostname from the configuration in the NodePingPeriodical

### PR DESCRIPTION
**Pull request by janheise**
## Description

To ensure that we always use the hostname that a user has specified.

favoring this over #17004 

/nocl

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: data-node/src/test/java/org/graylog/datanode/periodicals/NodePingPeriodicalTest.java [hunk 1]
```diff
@@ -81,4 +90,38 @@ void doRunWithRegister() throws NodeNotFoundException {
                 Mockito.any()
```
Comment (mpfz0r) [inline data-node/src/test/java/org/graylog/datanode/periodicals/NodePingPeriodicalTest.java:90]: If you change that to match "localhost" then the new test becomes redundant

**Review by mpfz0r — COMMENTED** [review]
one small comment. rest looks good

File: data-node/src/test/java/org/graylog/datanode/periodicals/NodePingPeriodicalTest.java (hunk 1 above)
Comment (janheise) [reply data-node/src/test/java/org/graylog/datanode/periodicals/NodePingPeriodicalTest.java]: thx. I made the test more meaningful and removed the new test.

**Review by mpfz0r — APPROVED** [review]

**Comment by mpfz0r** [comment]
@janheise while you're at it, you might take a look at https://github.com/Graylog2/graylog2-server/issues/17005 as well

**Pull request merged**